src/Settings/Default.hs

```diff
@@ -215,6 +215,52 @@ defaultBuilderArgs = mconcat
     , makeBuilderArgs
     , tarBuilderArgs ]
 
+-- | Disable some warnings in packages we use
+-- | https://github.com/ghc/ghc/blob/master/mk/warnings.mk#L46
+disableWarningArgsStage0 :: Args
+disableWarningArgsStage0 = stage Stage0 ? builder Ghc ? mconcat
+    [ package transformers ? append [ "-fno-warn-unused-matches", "-fno-warn-unused-imports" ]
+    , package terminfo ? append [ "-fno-warn-unused-imports" ] ]
+
+-- NOTE(izgzhen): The commented-out packages are NOT built by default with make either,
+-- so they are only here for consistency
+disableWarningArgsStage1 :: Args
+disableWarningArgsStage1 = notStage0 ? builder Ghc ? mconcat
+    [ package bytestring ? append [ "-Wno-inline-rule-shadowing" ]
+    , package haddock ? append [ "-Wno-unused-imports", "-Wno-deprecations" ]
+--    , package vector ? append [ arg "-Wno-unused-imports", arg "-Wwarn" ]
+    , package directory ? append [ "-Wno-unused-imports" ]
+    , package binary ? append [ "-Wno-deprecations" ]
+    , package haskeline ? append [ "-Wno-deprecations", "-Wno-unused-imports",
+                                 "-Wno-redundant-constraints",
+                                 "-Wno-simplifiable-class-constraints" ]
+    , package pretty ? append [ "-Wno-unused-imports" ]
+    , package primitive ? append [ "-Wno-unused-imports", "-Wno-deprecations" ]
+--    , package random ? append [ "-Wno-redundant-constraints" ]
+    , package terminfo ? append [ "-Wno-unused-imports" ]
+    , package xhtml ? append [ "-Wno-unused-imports", "-Wno-tabs" ]
+--    , package dphBase ? append [ "-Wwarn" ]
+--    , package dphPrimInterface ? append [ "-Wwarn"]
+--    , package dphPrimSeq ? append [ "-Wwarn"]
+--    , package dphPrimPar ? append [ "-Wwarn"]
+--    , package dphLiftedCommon ? append [ "-Wwarn"]
+    , package transformers ? append [ "-Wno-unused-matches", "-Wno-unused-imports",
+                                    "-Wno-redundant-constraints", "-Wno-orphans" ]
+    , package base ? append [ "-Wno-trustworthy-safe" ]
+    , package ghcPrim ? append [ "-Wno-trustworthy-safe" ]
+    , package win32 ? append [ "-Wno-trustworthy-safe" ] ]
+
+-- GhcLibExtraHcOpts += -Wno-deprecated-flags
+-- GhcBootLibExtraHcOpts += -fno-warn-deprecated-flags
+disableWarningArgsLibs :: Args
```

Comment:
not sure whether it would be profitable to have another combinator, since I found `isLibrary`'s implementation quite simple